src/main/java/gift/service/MemberService.java

```diff
@@ -0,0 +1,49 @@
+package gift.service;
+import static gift.dto.MemberDto.of;
+
+import gift.domain.Member;
+import gift.dto.MemberDto;
+import gift.exception.MemberAlreadyExistsException;
+import gift.exception.MemberNotExistsException;
+import gift.repository.MemberRepository;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MemberService {
+    private final MemberRepository memberRepository;
+
+    public MemberService(MemberRepository memberRepository) {
+        this.memberRepository = memberRepository;
+    }
+
+    public List<MemberDto> findAll() {
+        return memberRepository.findAll().stream()
+            .map(MemberDto::of)
+            .toList();
+    }
+
+    public MemberDto find(String email) {
+        Optional<Member> member =  memberRepository.findByEmail(email);
+        member.orElseThrow(MemberNotExistsException::new);
+        return of(member.get());
+    }
```

Comment:
옵셔녈 사용 좋습니다~ 이렇게 바꿀 수도 있겠네요.
```suggestion
    public MemberDto find(String email) {
        Member member = memberRepository.findByEmail(email)
                                        .orElseThrow(MemberNotExistsException::new);
        return MemberDto.of(member);
    }
```